### DISCUSSION TITLE
What is the difference between Core.ACPI from System.Power?

### DISCUSSION THREAD

**Discussion by Shlepsss** [CosmosOS/Cosmos#2553] [Q&A] [answered]
What is the difference between Core.ACPI from System.Power?

**Comment by ascpixi** [comment, marked as answer]
`Cosmos.Core.ACPI` contains the implementation for the ACPI standard. `Cosmos.System.Power` is more of an abstraction; it's only composed of two methods, currently: the `Reboot` and the `Shutdown` method. Both methods internally call `HAL.Power.ACPIShutdown` (which in turn call the method from `Cosmos.Core.ACPI`) and `HAL.Power.CPUReboot`.

This means that shutting down is the same as with ACPI; the rebooting, however, is done differently.
(keep in mind ACPI rebooting is not yet implemented)

**Reply by Shlepsss** [reply]
thanks!
